config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala

```diff
@@ -924,7 +924,7 @@ class ConfigSubstitutionTest extends TestUtils {
 
     @Test
     def substSelfReferenceIndirect() {
-        val obj = parseObject("""a=1, b=${a}, a=${b}""")
+        val obj = parseObject("""b=${a}, a=${b}""")
```

Comment:
This is getting complicated, so I filed a new bug just for this issue: https://github.com/typesafehub/config/issues/253